src/tequila/circuit/compiler.py

```diff
@@ -643,10 +593,20 @@ def compile_power_base(gate):
     """
     if not isinstance(gate, PowerGateImpl):
         return QCircuit.wrap_gate(gate)
-    power = gate.parameter
-    if gate.name in ['H', 'h', 'Hadamard', 'hadamard']:
-        return compile_h_power(gate=gate)
-    if gate.name == 'X':
+
+    if gate.is_controlled():
+        return QCircuit.wrap_gate(gate)
+
+    power = gate.power
+    if gate.name.lower() in ['h', 'hadamard']:
+        ### off by global phase of Exp[ pi power /2]
+        theta = power * numpy.pi
+
+        result = QCircuit()
+        result += Ry(angle=-numpy.pi / 4, target=gate.target)
+        result += Rz(angle=theta, target=gate.target)
+        result += Ry(angle=numpy.pi / 4, target=gate.target)
+    elif gate.name == 'X':
         ### off by global phase of Exp[ pi power /2]
         '''
         if we wanted to do it formally we would use the following
```

Comment:
Nice. Very clear. Global phase is fine. Maybe add "Global phases can be ignored. For controlled gates the global phase becomes a relative phase so this structure is respected in the compilation of controlled gates." or something like this